recipe/meta.yaml

```diff
@@ -183,7 +183,6 @@ outputs:
       run:
         - {{ pin_subpackage('libxgboost', exact=True) }}
         - {{ pin_subpackage('_r-xgboost-mutex', exact=True) }}
-        - llvm-openmp  # [osx]
```

Comment:
Why was this dropped? Also should `libgomp` be here too?

Reply:
`host` should suffice. `libgomp` is added to `host`.

Reply:
Am seeing lines like this in [the CI log]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=750342&view=logs&j=cacb33e2-4f35-58d8-c6f0-bc569fa20097&t=5d96f9ff-2d0d-52ea-cc90-ffadc2f5efed&l=839 ):

```
WARNING (libxgboost,bin/xgboost): Needed DSO lib/libgomp.so.1 found in ['_openmp_mutex']
WARNING (libxgboost,bin/xgboost): .. but ['_openmp_mutex'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

Are we sure these are not needed in `run`?

Reply:
Apologies if I was too hasty at marking as resolved! 

Note that the `r-xgboost` section, where this PR's proposed changes are located, is clean in [the CI logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=750342&view=logs&j=cacb33e2-4f35-58d8-c6f0-bc569fa20097&t=5d96f9ff-2d0d-52ea-cc90-ffadc2f5efed&l=2854):

```bash
   INFO: sysroot: '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/zone.tab', '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/tzdata.zi', '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/right/Zulu', '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/right/WET']'
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/R/lib/libR.so found in conda-forge::r-base-4.2.3-hb1b9b86_6
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-13.1.0-hfd8a6a1_0
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/libgomp.so.1 found in conda-forge::_openmp_mutex-4.5-2_gnu
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-13.1.0-he5830b7_0
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/ld-linux-x86-64.so.2 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
WARNING (r-xgboost): plugin library (R) package conda-forge::r-jsonlite-1.8.7-r42h57805ef_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-xgboost): plugin library (R) package conda-forge::r-magrittr-2.0.3-r42h57805ef_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-xgboost): plugin library (R) package conda-forge::r-matrix-1.6_0-r42h316c678_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-xgboost): plugin library (R) package conda-forge::r-data.table-1.14.8-r42h029312a_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

The warning highlighted in the previous comment is from the `libxgboost` output section. That is, it is an existing problem in the recipe, in a section that builds *prior to* - i.e., *independent of* - any of the PR changes.

My experience with R recipes is that, despite what the documentation may say, OpenMP should be declared in `host` and only there. Declaring it in other locations (e.g., `build` + `run`) leads to problems, specifically, final metadata (e.g., `llvm-openmp>=16`) being incompatible with globally pinned version (e.g., `llvm-openmp=15`), leading to builds that are unusable in downstream solves. Perhaps this doesn't translate to library builds (`libxgboost`), but it is my working practice for R packages.

If this is misguided, please elaborate (the current documentation on OpenMP linking is merely prescriptive without any explanation).

Reply:
Ok agree that `llvm-openmp` & `libgomp` should work via `run_exports` and shouldn't be needed in `run`

Also agree that `r-xgboost` looks ok and this is a `libxgboost` issue